CardanoSharp.Wallet/CIPs/CIP2/CoinSelectionUtility.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using CardanoSharp.Wallet.CIPs.CIP2.ChangeCreationStrategies;
+using CardanoSharp.Wallet.Models;
+using CardanoSharp.Wallet.Models.Transactions;
+using CardanoSharp.Wallet.TransactionBuilding;
+
+namespace CardanoSharp.Wallet.CIPs.CIP2
+{
+    public static class CoinSelectionUtility
+    {
+        public static CoinSelection UseLargestFirst(this TransactionBodyBuilder tbb, List<Utxo> utxos)
```

Comment:
By the time the consumer is working with a TransactionBodyBuilder they should have already determined all the inputs and outputs? At the point of coin selection they only have an idea of the pending Outputs and a collection of their wallet Utxos.

Reply:
> At the point of coin selection they only have an idea of the pending Outputs and a collection of their wallet Utxos.

This is correct. The point of the CoinSelection is to provide us with our inputs and potential change outputs.

Reply:
This may be useful
https://cips.cardano.org/cips/cip2/#parameters